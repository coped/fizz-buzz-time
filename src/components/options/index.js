import Options from "components/options/Options";

export { Options };
