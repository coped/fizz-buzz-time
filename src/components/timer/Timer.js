import React from "react";
import "./Timer.scss";

export default function Timer(props) {
  const timerOutput = (count) => {
    return formattedTime(count);
  };

  const formattedTime = (totalSeconds) => {
    const time = secondsToTime(totalSeconds);
    return (
      pad(time.hours, 1) +
      ":" +
      pad(time.minutes, 2) +
      ":" +
      pad(time.seconds, 2)
    );
  };

  const secondsToTime = (totalSeconds) => {
    const secondsIn = {
      anHour: 3600,
      aMinute: 60,
    };

    let hours = 0;
    let minutes = 0;
    let seconds = 0;
    if (totalSeconds >= secondsIn.anHour) {
      hours = Math.floor(totalSeconds / secondsIn.anHour);
      totalSeconds -= secondsIn.anHour * hours;
    }
    if (totalSeconds >= secondsIn.aMinute) {
      minutes = Math.floor(totalSeconds / secondsIn.aMinute);
      totalSeconds -= secondsIn.aMinute * minutes;
    }
    seconds = totalSeconds;
    return { hours, minutes, seconds };
  };

  const pad = (s, n) => {
    return s.toString().padStart(n, "0");
  };

  const fizzBuzzOutput = (fizz, buzz, count) => {
    let output = "";
    if (count !== 0) {
      if (count % fizz === 0) output += "Fizz";
      if (count % buzz === 0) output += "Buzz";
    }
    return output;
  };

  const { toggleShowTimer, startTimer, stopTimer, count, fizz, buzz } = props;

  return (
    <div id="Timer">
      <button
        id="to-options"
        onClick={toggleShowTimer}
        type="button"
        className="button"
      >
        {"< Set times"}
      </button>
      <div className="timer">
        <p className="description">Time Elapsed</p>
        <div className="timer-container">
          <p id="timer-output">{timerOutput(count)}</p>
        </div>
        <div className="timer-buttons">
          <button
            id="start-timer"
            onClick={startTimer}
            type="button"
            className="timer-button start"
          >
            Start
          </button>
          <button
            id="stop-timer"
            onClick={stopTimer}
            type="button"
            className="timer-button stop"
          >
            Stop / reset
          </button>
        </div>
        <p id="fizz-buzz-output" className="bold">
          {fizzBuzzOutput(fizz, buzz, count)}
        </p>
      </div>
    </div>
  );
}
