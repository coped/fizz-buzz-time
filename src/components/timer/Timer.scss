#Timer {
  $button-height: 53px;

  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: flex-start;

  .timer {
    display: flex;
    width: 100%;
    height: calc(100% - (#{$button-height} + 20px));
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  #to-options {
    height: $button-height;
    margin-top: 20px;
    margin-left: 20px;
  }

  #fizz-buzz-output {
    margin-top: 65px;
    height: 100px;
    font-size: 80px;
  }

  .timer-buttons {
    margin-top: 50px;
  }

  .timer-button {
    width: 200px;
    border: none;
    border-radius: 5px;
    color: white;
    padding: 15px;
    text-align: center;
    font-size: 16px;
    font-weight: bold;
    cursor: pointer;
  }

  .stop {
    background-color: #e66666;
  }

  .start {
    background-color: #5ccc87;
    margin-right: 30px;
  }

  .timer-container {
    margin-top: 50px;
    font-size: 30px;
    font-weight: bold;
    border: 2px solid #707070;
    border-radius: 5px;
    width: 640px;
    background-color: #f5f5f5;
    padding: 28px;
  }
}
