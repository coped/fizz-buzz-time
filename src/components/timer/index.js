import Timer from "components/timer/Timer"

export { Timer }