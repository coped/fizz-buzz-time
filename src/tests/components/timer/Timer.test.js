import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Timer } from "components/timer";

Enzyme.configure({ adapter: new Adapter() });

describe("Timer", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Timer />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // it("has a valid snapshot", () => {
  //   const component = renderer.create(<Timer />);
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
