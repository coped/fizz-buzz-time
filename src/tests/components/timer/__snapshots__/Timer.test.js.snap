// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer has a valid snapshot 1`] = `
<div
  id="Timer"
>
  <button
    className="button"
    id="to-options"
    type="button"
  >
    &lt; Set times
  </button>
  <div
    className="timer"
  >
    <p
      className="description"
    >
      Time Elapsed
    </p>
    <div
      className="timer-container"
    >
      <p
        id="timer-output"
      >
        0:00:00
      </p>
    </div>
    <div
      className="timer-buttons"
    >
      <button
        className="timer-button start"
        id="start-timer"
        type="button"
      >
        Start
      </button>
      <button
        className="timer-button stop"
        id="stop-timer"
        type="button"
      >
        Stop / reset
      </button>
    </div>
    <p
      className="bold"
      id="fizz-buzz-output"
    >
      
    </p>
  </div>
</div>
`;
